store/postgres/src/block_range.rs

```diff
@@ -87,6 +87,31 @@ impl ToSql<Range<Integer>, Pg> for BlockRange {
     }
 }
 
+#[derive(Constructor)]
+pub struct BlockBoundsMatchClause<'a> {
+    table_prefix: &'a str,
+    block: BlockNumber,
+}
+
+impl<'a> QueryFragment<Pg> for BlockBoundsMatchClause<'a> {
+    fn walk_ast(&self, mut out: AstPass<Pg>) -> QueryResult<()> {
+        out.unsafe_to_cache_prepared();
+
+        out.push_sql("lower(");
+        out.push_sql(self.table_prefix);
+        out.push_identifier(BLOCK_RANGE_COLUMN)?;
+        out.push_sql(") = ");
+        out.push_bind_param::<Integer, _>(&self.block)?;
+        out.push_sql(" or upper(");
+        out.push_sql(self.table_prefix);
+        out.push_identifier(BLOCK_RANGE_COLUMN)?;
+        out.push_sql(") = ");
+        out.push_bind_param::<Integer, _>(&self.block)?;
```

Comment:
By doing `lower(block_range) = $b or upper(block_range) = $b` you'll get both the old and the new entity version in your query result. It's worth a thought if the API should actually return an indication to users what change was made (whether the entity was inserted, updated, or deleted at that block) The old version will have block range `[x, $b)`, the new one will have `[$b,y)` If you ran two queries, one where you get the entities with `lower(block_range) = $b` and another one where you only select the `id` where `upper(block_range) = $b` you could piece that together from just that information and don't need to read the entire old version.

Also, when checking the upper bound of the block range it's better to use the (very unwieldy) `coalesce(upper(block_range), 2147483647)` since there is an index for that rather than `upper(block_range)`, but that's a minor detail for now.